### ISSUE TITLE
使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错

### ISSUE THREAD

**Issue by frankqianghe** [wujiyang/Face_Pytorch#43]
您好：
       使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况

**Comment by foocker** [comment]
so, you solved this problem?

**Comment by Ontheway361** [comment]
emm， 
for data  : face occlusion,  side face, ...
for align : face bbox, face align ...

**Comment by wujiyang** [comment]
lfw的结果高并说明不了什么，能在megaface上刷高分数才行，还有就是要看你使用的什么训练集

**Comment by smartparrot** [comment]
> lfw的结果高并说明不了什么，能在megaface上刷高分数才行，还有就是要看你使用的什么训练集

请问adgdb-30 cfp 测试集在哪下载呀？ 我用insightface代码对训练集做了对齐，然后从你提供的云盘下载了lfw cfp adgdb30测试集， 这样行吗？  我担心的是训练集对齐方式和测试集对齐方式不一致

**Comment by smartparrot** [comment]
do you did face alignment?

**Comment by ypw-lbj** [comment]
请问您使用预训练模型了吗？

**Comment by wolfworld6** [comment]
> 您好：
> 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况

hello，when i training，the lfw acc was as follows，how i can solve it？
![image](https://user-images.githubusercontent.com/18051979/102753325-ff7d5200-43a5-11eb-84b6-a4686d224db6.png)

**Comment by QuasarLight** [comment]
Maybe the lr you used is too big.&nbsp;What machine are you using？



---原始邮件---
发件人: "wolfworld6"<notifications@github.com&gt;
发送时间: 2020年12月21日(周一) 下午4:03
收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;;
抄送: "Subscribed"<subscribed@noreply.github.com&gt;;
主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43)




  
您好：
 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况
  
hello，when i training，the lfw acc was as follows，how i can solve it？

**Comment by wolfworld6** [comment]
> Maybe the lr you used is too big.&nbsp;What machine are you using？
> […](#)
> ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月21日(周一) 下午4:03 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "Subscribed"<subscribed@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) 您好： 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况 hello，when i training，the lfw acc was as follows，how i can solve it？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

tesla v100， however i changed the lr=0.01，0.001，it is still so

**Comment by QuasarLight** [comment]
did you turn the option shuffle in dataloader into true?&nbsp;



---原始邮件---
发件人: "wolfworld6"<notifications@github.com&gt;
发送时间: 2020年12月22日(周二) 中午11:36
收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;;
抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43)




  
Maybe the lr you used is too big.&nbsp;What machine are you using？
 …
 ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月21日(周一) 下午4:03 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "Subscribed"<subscribed@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) 您好： 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况 hello，when i training，the lfw acc was as follows，how i can solve it？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.
  
tesla v100， however i changed the lr=0.01，0.001，it is still so

**Comment by wolfworld6** [comment]
> did you turn the option shuffle in dataloader into true?&nbsp;
> […](#)
> ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月22日(周二) 中午11:36 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) Maybe the lr you used is too big.&nbsp;What machine are you using？ … ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月21日(周一) 下午4:03 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "Subscribed"<subscribed@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) 您好： 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况 hello，when i training，the lfw acc was as follows，how i can solve it？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe. tesla v100， however i changed the lr=0.01，0.001，it is still so — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

no,the shuffle value is false, trainloader = data.DataLoader(dataset, batch_size=64, shuffle=False, num_workers=2, drop_last=False),did you use lfw.py or lfw_2.py?

**Comment by QuasarLight** [comment]
try to turn the shuffle into true
---原始邮件---
发件人: "wolfworld6"<notifications@github.com&gt;
发送时间: 2020年12月22日(周二) 下午2:18
收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;;
抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43)




  
did you turn the option shuffle in dataloader into true?&nbsp;
 …
 ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月22日(周二) 中午11:36 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) Maybe the lr you used is too big.&nbsp;What machine are you using？ … ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月21日(周一) 下午4:03 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "Subscribed"<subscribed@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) 您好： 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况 hello，when i training，the lfw acc was as follows，how i can solve it？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe. tesla v100， however i changed the lr=0.01，0.001，it is still so — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
  
no,the shuffle value is false, trainloader = data.DataLoader(dataset, batch_size=64, shuffle=False, num_workers=2, drop_last=False),did you use lfw.py or lfw_2.py?

**Comment by wolfworld6** [comment]
> try to turn the shuffle into true ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月22日(周二) 下午2:18 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) did you turn the option shuffle in dataloader into true?&nbsp; … ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月22日(周二) 中午11:36 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "MaoCai"<1440896206@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) Maybe the lr you used is too big.&nbsp;What machine are you using？ … ---原始邮件--- 发件人: "wolfworld6"<notifications@github.com&gt; 发送时间: 2020年12月21日(周一) 下午4:03 收件人: "wujiyang/Face_Pytorch"<Face_Pytorch@noreply.github.com&gt;; 抄送: "Subscribed"<subscribed@noreply.github.com&gt;; 主题: Re: [wujiyang/Face_Pytorch] 使用您的代码，我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错 (#43) 您好： 使用您的代码我在lfw的得分可以达到99.6，但是实际应用场景会出现大批量的认错。为了不认错把阀值调高，又会出现认不出的情况。我初步分析是两个原因，一个是实际使用中有复杂的光照，还一个是训练数据集中大部分是欧美人脸，亚洲人脸很少。用欧美人脸训练出来的模型对亚洲人脸识别不好。请问您感觉还会有什么原因造成这种情况 hello，when i training，the lfw acc was as follows，how i can solve it？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe. tesla v100， however i changed the lr=0.01，0.001，it is still so — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. no,the shuffle value is false, trainloader = data.DataLoader(dataset, batch_size=64, shuffle=False, num_workers=2, drop_last=False),did you use lfw.py or lfw_2.py? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

sorry,my bad,i found my pairs.txt is wrong in the gdb mode.
